### PR TITLE
Placement and Intern APIs added

### LINKED ISSUE #11: Fetch User data API
Create an API which fetches the users data from the database.
It should work in the following fashion - 
```
method: GET
Successful : 200_OK
Unsuccessful : 404_BAD_REQUEST / 401_UNAUTHORIZED

Response : {
            "roll",
            "username",
            "name",
            "email",
            ...
}
```
The response should contain the required fields.
You can refer to [this](https://github.com/pclubiitk/Campus-Discuss-Backend) to check how it had been implemented in campus discuss.

### LINKED ISSUE #13: Fetch Intern Data for a given user.
You need to make an API which fetches the internships a user is eligible for. You may use session credentials for this, send a `GET` or a `POST` api (`GET` should do the work) and then get a list of all the internships he/she is eligible for by checking the dept, program, etc.

### LINKED ISSUE #14: Fetch Placement data for a given user.
You need to make an API which fetches the placements a user is eligible for. You may use session credentials for this, send a `GET` or a `POST` api (`GET` should do the work) and then get a list of all the placements he/she is eligible for by checking the dept, program, etc.

### PR DESCRIPTION
**Pull request by parinayc20**
The API has been tested and is working fine.
closes #14  
closes #13

### CODE REVIEW

File: user/views.py [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+    def get(self, request):
+        user = IsLoggedIn(request)
+        if user is not None:
+            try:
+                eligible_placement_ids = list()
+                all_placements = Placement.objects.all()
+                for placement in all_placements:
+                    if user.program in placement.eligible_programmes:
+                        if user.department in placement.eligible_branches:
+                            if user.batch in placement.eligible_batches:
```
Comment (namangup) [inline user/views.py]: Merge these if conditions in one using `and`

File: user/views.py [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+    def get(self, request):
+        user = IsLoggedIn(request)
+        if user is not None:
+            try:
+                eligible_intern_ids = list()
+                all_interns = Intern.objects.all()
+                for intern in all_interns:
+                    if user.program in intern.eligible_programmes:
+                        if user.department in intern.eligible_branches:
+                            if user.batch in intern.eligible_batches:
```
Comment (namangup) [inline user/views.py]: Same for this

**Review by namangup — CHANGES_REQUESTED** [review]
In README add the API documentation. Refer to campus-discuss repo for the format.

**Comment by parinayc20** [comment]
Made all the requested changes

**Pull request merged**